### PR TITLE
Standing PR with fixes detected by XSD.

### PR DESCRIPTION
**Pull request by jmchilton**

### CODE REVIEW

**Comment by jmchilton** [comment]
I'll put them out here in case people want to fight about them.

**Comment by bgruening** [comment]
Thanks John! Do you have plans to improve the XSD for Galaxy tools? I was trying to find someone that can teach the basics at least so we can all help to make it complete. I think this would also solve the documentation issue.

**Comment by jmchilton** [comment]
@bgruening https://github.com/galaxyproject/Galaxy-XSD/commits/master - A one man hackathon - I've added thousands lines in last 24 hours I think, it now validates all of Galaxy tools and the first 150 randomly chosen iuc tools. I'll keep working through it.

After I'm confident that it is validating IUC and Galaxy test tools - I'll swing back through and work on documentation - I've already started on it and I have a pretty clear plan how to generate markdown from the schema and replace https://wiki.galaxyproject.org/Admin/Tools/ToolConfigSyntax - though that work is still in a local branch.

**Comment by bgruening** [comment]
This is amazing! John tell me such things and I will try to help you! This is just wonderful. I was hoping to get this running with https://github.com/galaxyproject/galaxy/issues/2897 but so fast. Thanks a bunch!

File: tools/bcftools/bcftools_call.xml [hunk 1]
```diff
@@ -128,7 +128,7 @@ ${section.variants_only}
                         <when value="none">
                         </when>
                         <when value="trio">
-                            <param name="novel_rate" type="float" label="Novel Rate" default="1e-8,1e-9,1e-9" optional="true" help="likelihood of novel mutation for constrained trio calling, see man page for details" />
+                            <param name="novel_rate" type="float" label="Novel Rate" value="1e-8,1e-9,1e-9" optional="true" help="likelihood of novel mutation for constrained trio calling, see man page for details" />
```
Comment (jmchilton) [inline tools/bcftools/bcftools_call.xml]: Ping @bgruening - this is closer now but still not right. Do you know what happened here?

Comment (bgruening) [inline tools/bcftools/bcftools_call.xml]: This should be a text field. Afaik comma separated fields do not support commas separation. It should be 3 float values. Which we can either model with 3 float params or one text field.

**Comment by jmchilton** [comment]
Okay - as of now the XSD validates this entire branch. I'd rather not work on the bcftools 3 float value things - seems outside the scope of this. Anyone willing to take over this work and clean those up? Other than that though the changes in here seem pretty uncontroversial.

**Comment by bgruening** [comment]
I can review and take it over from Wednesday next week on. I also think this is uncontroversial and are great fixes. I'm not sure if we need to increase the build-number for a few of them.

Thanks John!

**Comment by martenson** [comment]
@jmchilton you switched to Atom 😨 ?

(looks good! :D 🎈 🎈 🎈 )

**Comment by jmchilton** [comment]
@martenson Just for today - it renders Markdown so beautifully.

**Comment by bgruening** [comment]
@erasche can you please look at the ART failure? It seems there is a min="101" but we have `          <param name="fragment_size" value="75" />` as test.

**Comment by bgruening** [comment]
Closed in favor of https://github.com/galaxyproject/tools-iuc/pull/939

**Pull request closed without merging**